### ISSUE TITLE
Test Polylang

### ISSUE THREAD

**Issue by jonathanbossenger** [WordPress/Learn#2284]
Front-end requirements:

* Language selector **does not show** as country flags. (Many countries speak multiple languages, so describing a language with a flag should be avoided.)
* Languages can be changed without needing to log in to one's WordPress.org account.
* Once a language is selected, it is applied globally.
* Content of the selected language is shown above non-translated content.
* English content is still shown where translations don't exist.

Back-end requirements:
* Translators do not have permission to create or publish new content.
* Translators have permission to both translate any content, and review content translated by others.
* A flow can be implemented where translations must be approved (reviewed) before published.
* Sensei content types can be translated in the same way as native post types.
* Taxonomies can be translated.
* Translated content is indexed and searchable by Jetpack Search.

**Comment by jonathanbossenger** [comment]
@cynthianorman feel free to leave your testing feedback here.

**Comment by cynthianorman** [comment]
Successfully installed on Learn WordPress local dev env
https://drive.google.com/file/d/1FhK_aov9Ut8Plex77k8LbDFTWdIGs90A/view?usp=drive_link

**Comment by cynthianorman** [comment]
Consider upgrading at some point to Pro
https://drive.google.com/file/d/1Jj_F4M59J4P52NTM1bk4bEKFO-WFLRDs/view?usp=drive_link

**Comment by cynthianorman** [comment]
We can select the languages we want to translate pages and posts to
https://drive.google.com/file/d/1npTZ3t9k9rWnaFMiVKHordb8YcKumMhF/view?usp=drive_link

Note that showing the flag in the front end is optional

**Comment by cynthianorman** [comment]
You can create translated menus
https://drive.google.com/file/d/1pvQYi1QSD5u3pTduH3G1K1rxpinK_P5v/view?usp=drive_link

**Comment by cynthianorman** [comment]
We assign a language to a page or post
https://drive.google.com/file/d/1QLN53RIfDCOAZlsqHBYE0NWbdLIJ82c-/view?usp=drive_link
https://drive.google.com/file/d/1y92lEZv_rpNSI6BZy_s34wCuoel8A8uD/view?usp=drive_link
https://drive.google.com/file/d/1HT_XT9a9XGh7g5w6mDBXNrMRO4AlUqNQ/view?usp=drive_link

Note, we are able to do the same thing with courses
https://drive.google.com/file/d/1VegkHlSONpCsF5Oh6DfQVu9f4pSDn6ex/view?usp=drive_link

**Comment by cynthianorman** [comment]
It's important to know that we essentially build a separate set of pages/posts/taxonomies and corresponding menu for each language. 

For this reason, we would need to create the copied English page/post and designate it as French (for example) otherwise I don't believe this will work out for us. In other words, the copied English page/post would act as a placeholder until a translator would contribute the translation.

https://drive.google.com/file/d/1ZLAMfGTc4nOa6Cl9Wdgo5BFpZEiBf4KR/view?usp=sharing

**Comment by jonathanbossenger** [comment]
@cynthianorman do you think you'd be able to update this issue with a summary of your findings, in a similar way as the [WPML test](https://github.com/WordPress/Learn/issues/2338#issuecomment-2010974864)? 

Edit, I've prepared the list in the comment below, so you just need to mark off the relevant items.

This will allow me to prepare a summary comparison table of our test findings for the training team meeting on Thursday.

**Comment by jonathanbossenger** [comment]
Front-end requirements:

- [x] Language selector does not show as country flags. (Many countries speak multiple languages, so describing a language with a flag should be avoided.)
- [x] Languages can be changed without needing to log in to one's WordPress.org account.
- [x] Once a language is selected, it is applied globally.
- [ ] Content of the selected language is shown above non-translated content.
- [x] English content is still shown where translations don't exist.

Back-end requirements:

- [x] Translators do not have permission to create or publish new content.
- [x] Translators have permission to both translate any content, and review content translated by others.
- [x] A flow can be implemented where translations must be approved (reviewed) before published.
- [x] Sensei content types can be translated in the same way as native post types.
- [x] Taxonomies can be translated.
- [x] Translated content is indexed and searchable by Jetpack Search.

**Comment by cynthianorman** [comment]
> @cynthianorman do you think you'd be able to update this issue with a summary of your findings, in a similar way as the [WPML test](https://github.com/WordPress/Learn/issues/2338#issuecomment-2010974864)?
> 
> Edit, I've prepared the list in the comment below, so you just need to mark off the relevant items.
> 
> This will allow me to prepare a summary comparison table of our test findings for the training team meeting on Thursday.

ok @jonathanbossenger done

**Comment by jonathanbossenger** [comment]
Perfect, thank you so much!

**Comment by jonathanbossenger** [comment]
@sebastienserre I have been testing PolyLang this week, and I have encountered a blocker that I hope you can help resolve.

Learn.WordPress.org uses Sensei LMS to manage our course content: https://wordpress.org/plugins/sensei-lms/

Sensei allows you to create course, and then create modules within the course and lessons within the modules:

![localhost_8888_wp-admin_post php_post=258770 action=edit](https://github.com/user-attachments/assets/82b94c72-ac48-48c7-8fbb-cf2c76d70492)

I am then able to navigate to individual lessons, and use the PolyLang feature to create different language versions of the individual lessons successfully:

![localhost_8888_wp-admin_post-new php_post_type=lesson](https://github.com/user-attachments/assets/428188ba-0cc7-4126-bd3e-6a5c25e743de)

However, as soon as I assign the language and link the translated version of a lesson to the English one, the relationship between the module and the lesson is broken. If I navigate back to the course edit screen, the translated lesson does not appear anymore in the module. 

![localhost_8888_wp-admin_post php_post=258770 action=edit (1)](https://github.com/user-attachments/assets/e78410a2-77a9-4ef0-b3a2-d6d5a5da7d29)

Are you able to assist in determining why this is happening?

**Comment by sebastienserre** [comment]
Hello @jonathanbossenger Here is a quick video where I show you What I'have found. I'm not used to use Sensei and at helpdesk we do not have so much request for this plugin, so I hope I'm matching your request.
https://www.loom.com/share/058116fd5b204318bf3d730f8c1b5b71

The "course" post meta seems to be synchronized between the same lesson in different language. This  is due to a WPML compatibility because Polylang is compatible with the WPML API to understand plugin & theme compatibility made for this plugin.
[WPML API in Polylang](https://polylang.pro/doc/wpml-api/)

To stop this synchronization, you can create a `wp-content/polylang/wpml-config.xml` file containing 
```
<wpml-config>
    <custom-fields>
        <custom-field action="ignore">_lesson_course</custom-field>
    </custom-fields>
</wpml-config>
```
If some others post_meta needs to be ignored, then `_lesson_course` is the meta name.
https://polylang.pro/doc/the-wpml-config-xml-file/

Of course, what is done thanks to the `wpml-config.xml` file can also be done in PHP if the wp.org maintainers prefer.

**Comment by jonathanbossenger** [comment]
Hey @sebastienserre, no problem at all; I'm not used to PolyLang, so we're all figuring this out together. 😁

> The "course" post meta seems to be synchronized between the same lesson in different language

That does seem to be correct, so I will test your suggestion and report back with my findings.

**Comment by sebastienserre** [comment]
The initial sync is made by the Sensei LMS wpml-config.xml available at their root https://plugins.trac.wordpress.org/browser/sensei-lms/trunk/wpml-config.xml

You can see lots of copied "custom fields"  some other post metas are sync and may "break" your test. Adding some lines with `ignore` instead of `copy` should help. In case of doubts, I'm available.

Like WordPress with a child theme, Polylang will read first the plugin root wpml-config.xml file, then the parent theme one (if exists), then the child theme one, then check in `mu-plugins` folder and finally `wp-content/polylang`

**Comment by jonathanbossenger** [comment]
Hello @sebastienserre. With the suggested wpml-config.xml config in place, I am still experiencing the problem.

I have created a screen recording of what I am seeing, I hope this will help you understand what I'm experiencing. 

https://github.com/user-attachments/assets/f5e7a81d-2162-4b30-87e3-88a6b99727c0

**Comment by jonathanbossenger** [comment]
Hi @sebastienserre I wanted to follow up here, have you been able to look into this?

Thanks

**Comment by sebastienserre** [comment]
Hello @jonathanbossenger 
I'm away a few days, I will test and answer quickly

**Comment by jonathanbossenger** [comment]
@sebastienserre not a problem thanks for letting me know, I just wanted to make sure you got the message.

**Comment by sebastienserre** [comment]
Hello @jonathanbossenger 

Here are 2 videos (yes I'm not ready to contribute to learn team, I have problems to click on the correct button :) )

https://www.loom.com/share/7f4737cdf22342698e9b7198190beea8
https://www.loom.com/share/3c861d48eeca4d569f7498fe0b96cdfc

I've set up a Twenty Twenty One child theme with the `wpml-config.xml` file on its root.

As you can see, I do not follow exactly the same steps you're following, but the result is I have a course translated in 2 languages.

If you do not have the behaviour I have, I think a conflict is occurring... Does it work with Twenty Twenty One ?

Where can I download the learn.wp.org theme to test it ?

**Comment by jonathanbossenger** [comment]
Ok, thanks @sebastienserre I will review the videos next week and check against the Learn theme.

**Comment by jonathanbossenger** [comment]
@sebastienserre thanks for the feedback. I can confirm that I can translate lessons the way you have shown.

However, I'm seeing a weird thing that happens when I try to create lessons inside a module in the Course edit screen, which I think may be a Sensei related issue. I'm going to see if I can get some help with this.

**Comment by jonathanbossenger** [comment]
@kaitohm it looks like we might have a workable solution here. I was able to test it on a clean Sensei environment, so I just need to test it on the learn.wordpress.org local environment (which I plan to do today) and will report back here once that's done.

**Comment by kaitohm** [comment]
Great news! Thanks for the update 👍

**Comment by jonathanbossenger** [comment]
@kaitohm Ok, here is a recording of the process of translating lesson content:

https://github.com/user-attachments/assets/c68e6685-df3e-411c-8dc2-02dae481fd6e

In short, Polylang allows us to translate lesson content using the regular Lesson editing user interface and retain all the relevant course/module/lesson structures. 

All we need to ensure is the following:

1. We include a `wpml-config.xml` file in the Learn child theme which contains a specific rule about the _lesson_course post meta field (I have this and can create the PR)
2. We create a specific user role for translation users who can edit/create lessons (I can work with @adamwoodnz on getting this done).

Therefore, I propose we move ahead with the next step, which would be to arrange for the code review of PolyLang to be included in the .org infrastructure in order to use it on Learn.

**Comment by kaitohm** [comment]
Thanks for the recording, @jonathanbossenger . This looks great! Please move forward with your suggestions above 👍

**Comment by jonathanbossenger** [comment]
@adamwoodnz can we start the process of having the PolyLang and PolyLang Pro code reviewed for inclusion into .org?

**Comment by sebastienserre** [comment]
@jonathanbossenger Polylang is included inside the Polylang Pro `vendor/wpsyntex`  folder. 
No need to have both plugins on the website. (On Polylang Pro activation, Polylang is automatically deactivated)

**Comment by jonathanbossenger** [comment]
Ah, thanks @sebastienserre. @adamwoodnz it looks like all we need to get reviewed then is [the plugin in the wordpress.org repo](https://wordpress.org/plugins/polylang/).

**Comment by adamwoodnz** [comment]
Sorry, which version are we planning to install on wordpress.org, Polylang or Polylang Pro?

**Comment by jonathanbossenger** [comment]
Hey @adamwoodnz, according to what @sebastienserre shared, we only need to test [Polylang](https://wordpress.org/plugins/polylang/).

**Comment by adamwoodnz** [comment]
> Polylang is included inside the Polylang Pro

Sounds like we'd need to install Pro, and it includes the base plugin

**Comment by jonathanbossenger** [comment]
Sorry @adamwoodnz I read the original message wrong. If we can test Polylang Pro please. I have a copy of the plugin code, let me know the best way to get it to you.

**Comment by sebastienserre** [comment]
We released Polylang Pro 3.6.5 last week to be OK with WP6.7. @jonathanbossenger you should be able to download it from your account at https:://polylang.pro/account.
Let me know if you encounter any difficulties

**Comment by adamwoodnz** [comment]
> We released Polylang Pro 3.6.5 last week to be OK with WP6.7. [@jonathanbossenger](https://github.com/jonathanbossenger) you should be able to download it from your account at https:://polylang.pro/account. Let me know if you encounter any difficulties

So given that we're installing the Pro version I wonder if we should add the code to this repository like we do for [Sensei Pro](https://github.com/WordPress/Learn/tree/trunk/wp-content/plugins/sensei-pro). If so you could download the code and open a PR @jonathanbossenger. Then we can review that.

This would mean it wouldn't be available for other .org sites, only Learn, which might be desirable. @dd32 what do you think?

**Comment by sebastienserre** [comment]
Hello

Selling Polylang Pro is the only way for us (8 persons) to live.
We're offering it to wp.org to make their learn website multilingual, but we disagree to offer it on a public GitHub Repo.

Thank you for your understanding.

**Comment by jonathanbossenger** [comment]
I'm happy to do that if that works best for everyone involved.

**Comment by sebastienserre** [comment]
Hello,

I'm going to see with Polylang's founder if it's reasonable to expose our Pro plugin (Unique surce of revenue for our company) in a public Github repository.

Regards

**Comment by jonathanbossenger** [comment]
@sebastienserre it might be helpful here to confirm if Learn actually needs the pro version of the plugin.

As far as I can tell, all we need to be able to do is create translated versions of courses, modules, and lessons with support for a language switcher. Correct me if I am wrong, but this is part of the core plugin available on the .org directory.

**Comment by sebastienserre** [comment]
@jonathanbossenger Site Editor, block editor is managed by Polylang pro, not Polylang. 
I've checked internally and this is OK to versioned Polylang Pro. If it's easier for every one, it's OK for us.

**Comment by adamwoodnz** [comment]
> block editor is managed by Polylang pro, not Polylang

Sorry, but to be clear; we use the block editor for Courses and Lessons, so does that mean we need Pro?

I didn't quite appreciate that we would be open sourcing your Pro code with this approach 🤦 Thanks for checking, and making this possible.

Dion is currently on leave, so I'd like to await his return and opinion on this.

**Comment by sebastienserre** [comment]
Yes you need Polylang Pro. 
In a WordPress world, the code is GPL. What we're selling is my job which is support :).
I spoke today with my manager (https://github.com/Chouby) who is the Polylang founder. There's no problem, Polylang Pro is already available on some public GH

**Comment by jonathanbossenger** [comment]
Thanks for checking and sharing the update @sebastienserre.

@adamwoodnz I will create the PR this week and ping you here once it's ready.

**Comment by dd32** [comment]
> This would mean it wouldn't be available for other .org sites, only Learn, which might be desirable. [@dd32](https://github.com/dd32) what do you think?

Learn is deployed into the same network that all other WordPress.org sites are in, so the plugin would be available to them, just not activated.

The main reason to include the Pro source in the GitHub repo is if you need it for local development sites of Learn, otherwise we can safely deploy it straight to WordPress.org without it going through this repo. That's why Sensei Pro is included AFAIK, because it was needed for local development and the free version didn't provide the same experience.

If you can share the code with me when you've got it, I can run it against the internal plugins tooling to ensure that no issues are flagged.

**Comment by dd32** [comment]
> If you can share the code with me when you've got it, I can run it against the internal plugins tooling to ensure that no issues are flagged.

Just noting that I've run the latest version of Polylang Pro 3.6.5 through the scanners I have access to, and nothing is being flagged as needing extra review, so I consider this good to deploy.

**Comment by adamwoodnz** [comment]
@jonathanbossenger if you could proceed with the PR I can add any necessary tooling changes for deployment, thanks all.

**Comment by jonathanbossenger** [comment]
Thanks, everyone. I'll create the PR and tag @adamwoodnz for the review.

**Comment by jonathanbossenger** [comment]
Adding a note that we need to add the wpml-config.xml to the wporg-learn-2024 theme with the following content, once the plugin is merged.

```
<wpml-config>
    <custom-fields>
        <custom-field action="ignore">_lesson_course</custom-field>
    </custom-fields>
</wpml-config>
```

**Comment by jonathanbossenger** [comment]
@adamwoodnz created #3049 to merge the Polylang Pro code.

I'll DM you the training team canvas with the logins.

**Comment by adamwoodnz** [comment]
> Adding a note that we need to add the wpml-config.xml to the wporg-learn-2024 theme with the following content, once the plugin is merged.
> 
> ```
> <wpml-config>
>     <custom-fields>
>         <custom-field action="ignore">_lesson_course</custom-field>
>     </custom-fields>
> </wpml-config>
> ```

Still need to do this, and test on https://learn.wordpress.org/test/

**Comment by adamwoodnz** [comment]
The plugin is activated on the test site with 2 languages: https://learn.wordpress.org/test/wp-admin/admin.php?page=mlang

Have a play and let me know what further code changes we need 🙂 

cc @kaitohm

**Comment by kaitohm** [comment]
Thanks, will look into this 👀

**Comment by jonathanbossenger** [comment]
Thanks @adamwoodnz 

@kaitohm when you're testing this, [this video](https://github.com/WordPress/Learn/issues/2284#issuecomment-2459070351) includes the process I followed to translate content.

I will also make a point to look into this later this week.

**Comment by jonathanbossenger** [comment]
@rjekic following on from our conversation in Slack, this is the issue where we were testing the implementation of PolyLang for Learn WordPress translations.

We have a test site at https://learn.wordpress.org/test/. You can login using your regular WordPress.org credentials. The test site data can be edited as you need.

The plugin is already installed, and I tested it [briefly here](https://github.com/WordPress/Learn/issues/2284#issuecomment-2516462608), but it would just be nice to have one more person test and confirm it will work for our requirements.

Once we confirm this will work, we can move forward with merging the code to the live site.

Let me know if you have any questions.

**Comment by rjekic** [comment]
@jonathanbossenger thank you for tagging me. I've already started testing Polylang on the local Learn environment, and I will soon give my feedback. 

> We have a test site at https://learn.wordpress.org/test/. You can login using your regular WordPress.org credentials. The test site data can be edited as you need.

Regarding this part, I've tried accessing the dashboard, but after logging in, it redirects me to the main Learn website. Can you check my role on the test site, or should I try to contact and ask someone from the meta team?

Thank you.

**Comment by jonathanbossenger** [comment]
🤦 My apologies @rjekic I completely forgot one needs to be added to the test site.

I've invited you, @devmuhib009, and @sumitsinghwp as admins.

**Comment by Jiwoon-Kim** [comment]
### Learn WordPress Multilingual Content Strategy Proposal

#### 1. Current Status & Challenges

The current multilingual support on `learn.wordpress.org` does not provide a unified experience across the site. Key issues include:

- **Mismatch Between UI and Content Language Settings:**  
  Changing the language via the top navigation (e.g., `?locale=ko_KR`) mainly affects the UI translation, but filtering or switching the language of content (courses, lessons) is limited or inconsistent.
    - The `/lessons/` section has a language filter, but `/courses/` does not.
    - Language settings are not fully synchronized between `wordpress.org` and `learn.wordpress.org`.
- **Lack of a Content Translation Workflow:**  
  There is no clear system to link translated courses to their originals or to manage users’ learning activities (such as course completion records) without duplication. This is a main reason for currently testing multilingual plugins like Polylang.
- ~**Desynchronization with Rosetta Sites:**~  
  ~Latest English news on `wordpress.org/news` is not immediately reflected on each language’s Rosetta site, causing time lags in information. Learn content may face similar issues.~

#### 2. Core Proposal: Separate UI Translation from Content Translation

To improve user experience and clarify contributor (translator) workflows, it is proposed to treat **site interface (UI) translation** and **content translation** as separate processes.

- **UI Translation:**  
  Control the interface language (menus, buttons, etc.) site-wide using parameters like `?locale=ko_KR`. This aligns with the current direction and should be expanded across the entire `wordpress.org` ecosystem.
- **Content Language:**  
  Regardless of UI language settings, users should be able to access content in their preferred language. For example, a developer using the Korean UI may want to take a course originally in English.

#### 3. Dual-Track Content Translation Strategy

To satisfy both accessibility and quality, the following two translation tracks should be offered simultaneously:

##### Track A: AI-Powered Instant Translation (Accessibility-Focused)

- **Goal:**  
  Ensure all users can access the latest content without language barriers.
- **Implementation:**
    1. Add an **"AI Translate" toggle or button** to all course and lesson pages.
    2. When clicked, display real-time translations using AI translation APIs such as Google Translate or DeepL.
    3. (Optional) Allow advanced users to register their own API keys for higher-quality translations.
- **Expected Outcome:**  
  Users in languages with limited translation resources can keep up with the latest information and learning materials.

##### Track B: Contributor-Driven Manual Localization (Quality-Focused)

- **Goal:**  
  Provide high-quality, culturally and contextually localized content for each language community.
- **Implementation:**
    1. Manually translated content is created as a separate post (course/lesson) with its own unique URL.
    2. Clearly link original and translated content, similar to Wikipedia’s **Interlanguage Links**. For example, display at the top of a course page: “This course is also available in: [Korean], [Español], ...”
    3. This approach naturally prevents duplicate course completions in user profiles (e.g., 'Introduction to Block Themes (English)' and '블록 테마 입문 (Korean)' are tracked as separate courses).
- **Expected Outcome:**  
  Maximizes learning effectiveness with officially reviewed, high-quality materials and clarifies the role of localization contributors.

#### 4. Conclusion & Expected Benefits

This proposal offers a more flexible and scalable multilingual strategy that does not conflict with current WordPress team considerations regarding multilingual plugins.

- **For Users:**  
  Freely choose UI and content languages, access any content instantly via AI, or select high-quality localized content for learning.
- **For Contributors:**  
  Clear goals for translation contributions (rapid information delivery vs. high-quality localization) and the ability to focus on content creation without complex technical linkage.
- **For the Platform:**  
  Reduced technical complexity (e.g., course history synchronization) and an ecosystem that respects the autonomy of each language community.

It is hoped that this proposal will contribute to discussions on the global expansion of `learn.wordpress.org`.

**Comment by Jiwoon-Kim** [comment]
https://github.com/WordPress/wordpress.org/issues/499

**Comment by Jiwoon-Kim** [comment]
I was wondering — is improving compatibility between Sensei and Polylang one of the main priorities or challenges for Learn WordPress at the moment?

- https://github.com/Automattic/sensei/issues/7604
- https://github.com/Automattic/sensei/issues/7563
- https://github.com/Automattic/sensei/pull/5855

**Comment by Jiwoon-Kim** [comment]
Even if we keep using query string parameters like
`https://learn.wordpress.org/lessons/?language[]=ko_KR&language[]=ja`
to filter content lists for now,

I believe **course permalinks should still follow a clean, localized URL structure like**:

`/ko/course/course-title/`
`/ja/course/course-title/`

This would improve clarity, consistency, and SEO for individual course pages, regardless of how content lists are filtered.

---

I think using /course/LOCALE/course-title/ would be a cleaner and more scalable permalink structure for localized courses.

**Comment by rjekic** [comment]
@Jiwoon-Kim 

Regarding this part of he comment above
>I think using /course/LOCALE/course-title/ would be a cleaner and more scalable permalink structure for localized courses.

This is not the option available in Polylang settings. As @sebastienserre already pointed out on Slack, this requires some custom coding and the maintenance workflow. Also, be aware that the course is a custom post type, like lesson, lesson-path, and tutorial. If we keep this logic you proposed, then, when we use lessons for example, we would have this - `/lesson/LOCALE/lesson-title/`.

I think we should keep the default Polylang option `/LOCALE/course/course-title/` and just focus on interaction and possible issues in manipulating content.

**Comment by Jiwoon-Kim** [comment]
@rjekic The reason I proposed two different permalink structures is because I’m not fully familiar with Polylang’s database structure — and since permalinks often reflect that structure indirectly, I wanted to explore what’s realistically possible.

Now that the limitations are clearer, I think we can consider discussing with the Meta team how to implement the `/LOCALE/course/course-title/` format. That seems like the most feasible direction from here.

**Comment by Jiwoon-Kim** [comment]
Using separate multisite subdirectories per locale might be too resource-intensive to manage effectively. Instead, it would make more sense to approach multilingual content through custom fields or custom post types (CPTs), which would allow for more flexible and scalable handling within a single site environment.

**Comment by arnasd** [comment]
### Missing translations

1. Translations for Sensei LMS plugin is missing in lots of places when switched to "Lithuanian" language. Marked in red color where translations are not there, everything is still in English.
<img width="2510" height="1748" alt="Image" src="https://github.com/user-attachments/assets/b79c332c-db13-4c22-a9df-da2ad3e2a848" />
Course link: https://learn.wordpress.org/test/lt/course/vertimo-testavimas/
2. Courses main page is not translated
<img width="1185" height="474" alt="Image" src="https://github.com/user-attachments/assets/7671f9b6-638d-49d4-976c-081282718b3e" />
3. It says "1 lesson" and "in-progress", no translation
<img width="565" height="170" alt="Image" src="https://github.com/user-attachments/assets/59c4d18e-3a51-47fa-91c2-16f332ea3de0" />

### Bugs

1.  There are 2 places where I can set the language, however only one should be there (the upper one in the header menu), as the "course language settings" are not working and do not change the language (shows that there are no courses too)

<img width="2488" height="1480" alt="Image" src="https://github.com/user-attachments/assets/e55260f8-0157-4284-9d8c-281d3ddf3ace" />

2. Global language selection is not visible in homepage https://learn.wordpress.org/test/

3. From https://learn.wordpress.org/test/lt/courses/ if I click on "Internetiniai seminarai" (online workshops) it just goes to https://learn.wordpress.org/test/online-workshops/ (without translation LOCALE and everything is reverted to English). Works correct if it's not expected to have a translation page

4. from https://learn.wordpress.org/test/lt/lessons/ if I click on "my courses" it reverts back to English selection (it is possible to choose the language in dropdown and it is set to English instead of Lithuanian as it was before) (https://learn.wordpress.org/test/my-courses/)

If from "my courses" I select "Lithuanian", I get "index" page without any information. So it seems that the issue is this page is missing a translation.
<img width="1243" height="573" alt="Image" src="https://github.com/user-attachments/assets/1e87ae8d-06cc-4a0b-bfbf-e66b6e77b897" />

✅  CONFIRMED - if I create a translated page, it redirected to the translated page. **But I still think it shouldn't go to "index" page, but actually load English (default version) instead.**



### Additional information

To create a new language for existing course/lesson - it is possible to now choose it from here to make things faster:

<img width="2518" height="1478" alt="Image" src="https://github.com/user-attachments/assets/5c06598e-023e-4900-a1bc-64f7cb40cf92" />

**Comment by arnasd** [comment]
Linking issues:

Issue for Sensei LMS translations -> https://github.com/WordPress/Learn/issues/3276
Duplicate language selectors -> https://github.com/WordPress/Learn/issues/3277

**Comment by sebastienserre** [comment]
For missing translations, the plugin must be translated in your locale at translate.wordpress.org:
For example, in French fr_FR, the translations of "0 of 1lessons completed (0%)" is translatable 
https://translate.wordpress.org/projects/wp-plugins/sensei-lms/stable/fr/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=13101190&filters%5Btranslation_id%5D=110919351

This is the case for all strings available from your theme & plugins templates

**Comment by sebastienserre** [comment]
The navigation block needs to be translated. Here, in the Italian test, you seem to use an English navigation block. 
Here is the official doc on Polylang Pro with Site Editor: https://polylang.pro/documentation/support/guides/site-editor/

**Comment by sebastienserre** [comment]
Hello @jonathanbossenger & @rjekic 
After our quick talk at WCEU in Krakow, do not hesitate to ping if you need some help with Polylang Pro.

**Comment by rjekic** [comment]
Hello @sebastienserre and @jonathanbossenger,

I finally (re)started working on this problem, and I think I found the solution.

The starting point was activating Polylang Pro in the Dashboard and labeling articles as **English (en_US),** and their corresponding translations as **Italian (it_IT)**.

## The expected behavior 

After all the previous work and settings, everything should work out-of-the-box, but it didn't. When I tried to change the language to Italian using the language switcher in the admin bar (global, shown when the user is logged in), the URL got the correct link (the `/?locale=it_IT` was correctly appended to the end of the URL), most of the page content was changed, but the courses and lessons did not get the locale info.

## The problem

Two systems independently decide what "the current language" is, and neither knows about the other:

- **Locale Detection** owns the **interface** locale. `Detector::set_locale()` resolves it from `?locale=`, then the `wporg_locale` cookie, then the browser's `Accept-Language` header, stores the result in that cookie for a year, and
  returns it from both the `locale` and `pre_determine_locale` filters
  (`locale-detection.php:26-27`, default priority 10).
- **Polylang** owns which **content** is served, taken from the language directory in the URL (`/it/`), and returns that language's locale from the `locale` filter (`frontend-filters.php:23`, also priority 10).

Same hook, same priority, so the winner is decided by load order - and Polylang forces itself to load first (`make_polylang_first()`), which means Locale Detection's filter registers later and overwrites it. **The cookie ends up deciding the interface while the URL decides the content, with nothing keeping the two equal.** That produced:

1. **The switcher didn't change the content.** Picking a language in the global header set the cookie and the interface strings, but Polylang kept serving the English page, because Polylang only reads the URL.
2. **The content didn't change the interface.** Opening `/it/` served Italian content inside an English interface, with `<html lang>`, date formats, and every translated string coming from the cookie's locale.
3. **A first visit could be mismatched with no action at all** - `Accept-Language` alone was enough to put an Italian interface around English content.
4. **The mismatch was baked into the page**, not just cosmetic: Polylang's OLT manager loads all text domains once, early (on `plugins_loaded`), from whatever `get_locale()` returned at that moment.
5. **Overlapping and missing controls**: Polylang's own admin-bar language node competing with the global header's locale switcher, and logged-out visitors with no switcher at all (the global header's is an admin-bar node, and its assets are only enqueued when the admin bar is showing).

## The solution (PR https://github.com/WordPress/Learn/pull/3615)

One rule: **the content language in the URL is the single source of truth for the whole request**, and anything that switches the interface is turned into a move to the matching content. The four parts below implement that - **(1)** makes the interface follow the content, **(2)** makes a switch move the content, and **(3)** and **(4)** leave exactly one control, in a place logged-out visitors can reach.

## What the mu-plugin does

1. **Interface follows content** - on the front end, it forces WordPress's `locale` / `pre_determine_locale` to the current Polylang language's locale (priority 99, so it wins over both `locale-detection` @10 and Polylang @20). `wp-admin` is left to Polylang.
2. **Content follows the switcher** - when a request carries `?locale=xx_XX` (what the interface switcher sets), it redirects to the same view in the matching Polylang language, using Polylang's own `get_translation_url()` (handles singular, pages, archives, taxonomies, home). Falls back to the language home when the view has no translation.
3. **No language controls in the admin bar** - Polylang's own language filter node (the one that sits right after "New") is suppressed via the `pll_admin_languages_filter` filter.
4. **Switcher next to the menu** - for logged-out visitors, a submenu of Polylang's languages, labelled with the current one, is appended to the last position of the theme's local navigation bar via `block_core_navigation_render_inner_blocks`. Its items link to the current URL with `?locale=xx_XX`, so `locale-detection` stores the choice in the `wporg_locale` cookie and (2) moves the visitor to that language's content. A translation icon (Gutenberg's `language` icon, the glyph the admin bar switcher showed via `dashicons-translation`) sits to the left of the label. A submenu's label is a block attribute and so can only be text, so the icon is inserted into the rendered markup on `render_block_core/navigation-submenu`, with a couple of alignment rules printed inline. It is built out of core's own `core/navigation-submenu` / `core/navigation-link` blocks, so it inherits the navigation's styling, submenu behaviour and mobile overlay with no extra CSS or JS. Only navigations carrying a `menuSlug` attribute are touched — that is the theme's local nav bar, including the always-collapsed copy the `wporg/local-navigation-bar` block renders for narrow screens; the global header and footer navigations are left alone. Logged-in users don't get it (they are assumed to use the global header's own locale switcher); drop the `is_user_logged_in()` check in `append_language_switcher_to_navigation()` to show it to everyone.

## Important note:

This solution is created as a **mu-plugin** with the big help from the AI assistant. I thoroughly checked the plugin and tried to keep the codebase as small as possible, maintaining a viable solution that does not break. I am not sure that using this as a **mu-plugin** is advised, but I could not reliably add it to the theme functions file because of the init process timing; maybe @sebastienserre can consider adding this to the Polylang Pro plugin directly, to be used when this situation occurs - when the Locale Detection plugin is used in parallel.

@jonathanbossenger and @sebastienserre 

Please review the proposed solution, and let's continue working on this. Sorry that this task took so long. 

cc @devmuhib009 @sumitsinghwp
